app/assets/javascripts/angular/components/organizationSignatures/organizationSignatures.component.js

```diff
@@ -0,0 +1,88 @@
+import template from './organizationSignatures.html';
+import './organizationSignatures.scss';
+
+angular.module('missionhubApp').component('organizationSignatures', {
+    bindings: {
+        orgId: '<',
+    },
+    template: template,
+    controller: organizationSignaturesController,
+});
+
+function organizationSignaturesController(
+    httpProxy,
+    $scope,
+    $window,
+    envService,
+    $http,
+) {
+    this.signatures = [];
+    this.signatureData = [];
+    this.exportLink = '';
+    this.itemsPerPage = 10;
+
+    const getSignatures = (orgId, searchText) => {
+        const params = {
+            include: [],
+            ...(searchText ? { 'filters[q]': searchText } : {}),
+        };
+
+        return httpProxy.get(`/organizations/${orgId}/signatures`, params, {
+            errorMessage: 'error.messages.surveys.loadQuestions',
+        });
+    };
+
+    const prepareData = data => {
+        return data.map(data => {
+            return {
+                first_name: data.first_name,
+                last_name: data.last_name,
+                updated_at: data.updated_at,
+                code_of_conduct: data.code_of_conduct,
+                statement_of_faith: data.statement_of_faith,
+            };
+        });
+
+        return data;
+    };
+
+    const buildExportCsvLink = searchText => {
+        const params = {
+            access_token: $http.defaults.headers.common.Authorization.slice(7),
+            include: '',
+            format: 'csv',
+            ...(searchText ? { 'filters[q]': searchText } : {}),
+        };
+
+        const query = Object.keys(params).reduce((accumulator, key, index) => {
+            return (
+                accumulator +
+                '&' +
+                encodeURIComponent(key) +
+                '=' +
+                encodeURIComponent(params[key])
+            );
+        }, '');
+
+        return (
+            envService.read('apiUrl') +
+            `/organizations/${this.orgId}/signatures?${query}`
+        );
+    };
+
+    const loadData = async searchText => {
+        const { data } = await getSignatures(this.orgId, searchText);
+        this.signatureData = prepareData(data);
+        this.signatures = [].concat(prepareData(data));
```

Comment:
I'm a little confused why both `signatureData` and `signatures` exist. I get that it's something to do with the table. Calling `prepareData` twice might not be necessary.
```suggestion
        this.signatures = [...prepareData(data)];
```
would do the same thing I think if you like that any better.

Reply:
yah so it looks like they mutate the main array of data when sorting, and filtering etc, so the duplicate is their "referenced copy" they can always revert to etc.  They re built the way this table system works in new angualr, vue, and react, but in the angualr1 its still like this.